api/src/main/java/org/ehcache/spi/ServiceProvider.java

```diff
@@ -23,17 +23,18 @@
  * This acts as a repository for {@link Service} instances, that can be used to
  * look them up by type.
  *
+ * @param <T> A bound on service types this provider can return
  * @author Alex Snaps
  */
-public interface ServiceProvider {
+public interface ServiceProvider<T extends Service> {
```

Comment:
There are a number of `ServiceProvider` uses to which the generic type parameter was not added.  Oversight?  Intentional?